File: HASKELL/Even_Odd.hs
--perform even odd in haskell

--using xmod2==0
isEven:: Int-> Bool
isEven x|x `mod`2==0=True
       |otherwise =False
main = do
       putStrLn"Enter a number to check it is even or not"
       ain <- getLine
       let a = (read ain::Int)
       print(isEven a)



--using xmod2==1
isOdd:: Int->Bool
isOdd x|x `mod`2==1=True
        |otherwise =False
main = do
       putStrLn"Enter the number to check it is odd or not"
       ain <-getLine
       let a=(read ain::Int)
       print(isOdd a)
     

File: PROLOG/even_odd.pl
sum([],[0,0]).
iseven(N) :-0 is mod(N,2). 
sum([H|T],[N1,N2]):- 
             sum(T,[N3,N4]),
             ( iseven(H) 

              ->  N1 is N3+H, N2 is N4
              ;   N2 is N4+H, N1 is N3

             )
             
             
%For output:-
%?- sum([5,4,9,8,1,7], L).
